template/addons/next-auth/next-auth.d.ts

```diff
@@ -0,0 +1,12 @@
+import { DefaultSession } from "next-auth";
+
+declare module "next-auth" {
+  /**
+   * Returned by `useSession`, `getSession` and received as a prop on the `SessionProvider` React Context
+   */
+  interface Session {
+    user: {
+      id?: string | null;
```

Comment:
```suggestion
      id: string;
```

Since you are enforcing it, this should always be defined now.


Make sure this `.d.ts` file is `include`d in the `tsconfig.json` to be picked up https://github.com/t3-oss/create-t3-app/blob/main/template/base/tsconfig.json#L19